index.d.ts

```diff
@@ -1,66 +1,257 @@
+
+	/**
+	 * Removes the widget from the page.
+	 * @param widgetId - The ID of the widget.
+	 */
+	remove(widgetId: WidgetId): void
+
+	/**
+	 * Obtains the widget's response using its widgetId.
+	 * @param widgetId - The ID of the widget.
+	 */
+	getResponse(widgetId: WidgetId): string
+
+	/**
+	 * Checks if the widget has expired.
+	 * @param widgetId - The ID of the widget.
+	 */
+	isExpired(widgetId: WidgetId): boolean
 }
 
-export interface TurnstileOptions {
-	sitekey: string;
-	action?: string;
-	cData?: string;
-	callback?: (token: string) => void;
-	"error-callback"?: () => void;
-	"expired-callback"?: (token: string) => void;
-	"timeout-callback"?: () => void;
-	"before-interactive-callback"?: () => void;
-	"after-interactive-callback"?: () => void;
-	"unsupported-callback"?: () => void;
-	theme?: "light" | "dark" | "auto"; // defaults to "auto"
-	language?: SupportedLanguages | "auto"; // defaults to "auto"
-	tabindex?: number; // defaults to 0
-	"response-field"?: boolean; // defaults to true
-	"response-field-name"?: string; // defaults to "cf-turnstile-response"
-	size?: "normal" | "invisible" | "compact"; // defaults to "normal"
-	retry?: "auto" | "never"; // defaults to "auto"
-	"retry-interval"?: number; // up to 15m (900_000) in ms, defaults to 8s
-	"refresh-expired"?: "auto" | "manual" | "never"; // defaults to "auto"
-	appearance?: "always" | "execute" | "interaction-only";
-	execution?: "render" | "execute";
-
-	chlPageData?: string; // ENTERPRISE ONLY - undocumented
+export type TurnstileOptions = RenderOptions
+
+/**
+ * Interface for Turnstile rendering parameters.
+ */
+export interface RenderOptions {
+	/**
+	 * Every widget has a sitekey. This sitekey is associated with the corresponding
+	 * widget configuration and is created upon the widget creation.
+	 * - Data Attribute - `data-sitekey`
+	 */
+	sitekey: string
+
+	/**
+	 * A customer value that can be used to differentiate widgets under the same
+	 * sitekey in analytics and which is returned upon validation. This can only
+	 * contain up to 32 alphanumeric characters including `_` and `-`.
+	 * - Data Attribute - `data-action`
+	 */
+	action?: string
+
+	/**
+	 * A customer payload that can be used to attach customer data to the challenge
+	 * throughout its issuance and which is returned upon validation. This can only
+	 * contain up to 255 alphanumeric characters including `_` and `-`.
+	 * - Data Attribute - `data-cdata`
+	 */
+	cData?: string
+
+	/**
+	 * Callback function invoked upon successful challenge completion.
+	 * - Data Attribute - `data-callback`
+	 * @param token - The token passed upon successful challenge.
+	 */
+	callback: (token: string) => void
+
+	/**
+	 * Callback invoked when there is an error (e.g., network error, challenge failed).
+	 * - Data Attribute - `data-error-callback`
+	 * @see [Client-side errors](https://developers.cloudflare.com/turnstile/reference/client-side-errors)
+	 */
+	'error-callback'?: () => void
```

Comment:
`error-callback` recently started to take a string as first argument (https://developers.cloudflare.com/turnstile/reference/client-side-errors)